HHS/Head-Start-TTADP
# [TTAHUB-1094] [TTAHUB-1135] Program Specialist email notifications

## Description of change

* The enum `enum_MailerLogs_action` was migrated to add two new fields: `granteeReportApproved` and `granteeReportApprovedDigest`.
* New email templates were added.
* The digest templates were updated to include new possible job name: `granteeReportApprovedDigest`.
* `granteeReportApprovedDigest` added to daily, weekly, monthly crons.
* `UserSettings` is migrated to include a new setting key: `emailWhenGranteeReportApprovedProgramSpecialist`
* The email preferences page now has an option for program specialists to receive emails when their grantee has an approved report.

<img width="647" alt="Screen Shot 2022-12-06 at 12 56 35 PM" src="https://user-images.githubusercontent.com/17074357/206010147-22b3c9cf-1219-4c0a-aae2-5f0d29c48918.png">

If an AR is approved which has more than one recipient where each share the same grant and program specialist (above), the subject of the email will include the names of each recipient:

<img width="974" alt="Screen Shot 2022-12-06 at 12 57 01 PM" src="https://user-images.githubusercontent.com/17074357/206010393-5d49c2cb-9b50-4c1a-9a84-99305ae1744a.png">

<img width="552" alt="Screen Shot 2022-12-06 at 12 57 07 PM" src="https://user-images.githubusercontent.com/17074357/206010496-5e3d5ae3-eaa8-4dda-a8d4-bae6d840e5b5.png">

(that AR display name above is a link now)

## How to test

*some unrelated tests were added to meet coverage threshold*


Create some known recipients on some known grants with yourself defined as the program specialist.

```sql
INSERT INTO "Recipients"
        ("id", "name", "createdAt", "updatedAt")
        VALUES (1010101, 'ABC Recipient', current_timestamp, current_timestamp);
INSERT INTO "Grants"
        ("id", "number", "recipientId", "regionId", "programSpecialistName", "programSpecialistEmail", "createdAt", "updatedAt", "status")
        VALUES (2020202, '123foobar456', 1010101, 1, 'Hello Specialist', 'jon.pyers@adhocteam.us', current_timestamp, current_timestamp, 'Active');
INSERT INTO "Recipients"
        ("id", "name", "createdAt", "updatedAt")
        VALUES (3030303, 'DEF Recipient', current_timestamp, current_timestamp);
INSERT INTO "Grants"
        ("id", "number", "recipientId", "regionId", "programSpecialistName", "programSpecialistEmail", "createdAt", "updatedAt", "status")
        VALUES (4040404, '123bazboo456', 3030303, 1, 'Hello Specialist', 'jon.pyers@adhocteam.us', current_timestamp, current_timestamp, 'Active');
```

Ensure your email is verified and you've opted into the new program-specialist-only email preference.

Create an AR and use those recipients, set yourself as the approver, and approve it. You should receive an email like the one shown above.

Additionally, ensure that the emails sent to you show up in the `MailerLogs` table.

The `down` migration should revert `enum_MailerLogs_action` to its original state, removing all mail logs that used the new enums.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1094
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1135


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
